tests/test_apdb.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of analysis_ap.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import unittest
+import tempfile
+
+import lsst.utils.tests
+from lsst.analysis.ap.apdb import ApdbSqliteQuery
+
+from lsst.obs.decam import DarkEnergyCamera
+import lsst.daf.butler
+
+
+class TestApdbSqlite(lsst.utils.tests.TestCase):
+    def setUp(self):
+        datadir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data/")
+        apdb_file = os.path.join(datadir, "apdb.sqlite3")
+
+        # TODO: only necessary until we can get detector/visit from APDB
+        path = tempfile.TemporaryDirectory()
+        lsst.daf.butler.Butler.makeRepo(path.name)
+        butler = lsst.daf.butler.Butler(path.name, writeable=True)
+        DarkEnergyCamera().register(butler.registry, update=True)
+
+        self.apdb = ApdbSqliteQuery(apdb_file, butler=butler, instrument="DECam")
+
+    def test_load_sources(self):
+        sources = self.apdb.load_sources()
```

Comment:
I take it that `load_sources()`, `load_objects()`, and `load_force_sources()` are deterministic functions, and that the 0th element in the returned data will be consistent, yeah?

Reply:
Yes, they all have `ORDER BY` statements, and the small sqlite test file is fixed (and if I update it, I'll update the tests).